### PR TITLE
riscv: Implement setjmp() and longjmp() for RISC-V

### PR DESCRIPTION
**Pull request by maroueneboubakri**
- Add jmp buffer size and type for RISC-V in setjmp.h
- Implement `setjmp()` and `longjmp()` in setjmp_rv.S

### CODE REVIEW

File: lib/libutils/isoc/arch/riscv/setjmp_rv.S [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#define STR       sw
+#define LDR       lw
+#define REGOFF(x) ((x) * 4)
+#else
+#define STR       sd
+#define LDR       ld
+#define REGOFF(x) ((x) * 8)
+#endif /*RV32*/
+
+// int setjmp (jmp_buf)
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/riscv/setjmp_rv.S:17]: Please use `/* */` styled comments.

File: lib/libutils/isoc/arch/riscv/setjmp_rv.S [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	.global	setjmp
+	.type	setjmp, %function
+setjmp:
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/riscv/setjmp_rv.S:20]: Please make this:
`FUNC setjmp , :` 
and drop the attributes above.

Same below for `longjmp()`.

File: lib/libutils/isoc/arch/riscv/setjmp_rv.S [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	STR	ra, (sp)
+	mov	x29, sp
+	addi	a0, a0, REGOFF(12)
+	jal	ftrace_setjmp
+	LDR	ra, (sp)
+	addi	sp, sp, 16
+#endif
+	li 	a0, 0
+	ret
+	.size	setjmp, .-setjmp
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/riscv/setjmp_rv.S:46]: `END_FUNC setjmp`, same below with `longjmp()`

File: lib/libutils/isoc/arch/riscv/setjmp_rv.S [hunk 4]
```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/arch/riscv/setjmp_rv.S:1]: Any particular reason why this isn't `BSD-2-Clause`?

**Review by jenswi-linaro — COMMENTED** [review]
You can drop the `lib: ` prefix if you like, it's pretty clear what `libutils: ` means.

**Comment by maroueneboubakri** [comment]
All comments have been addressed.

**Comment by maroueneboubakri** [comment]
> You can drop the `lib: ` prefix if you like, it's pretty clear what `libutils: ` means.

Done.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by maroueneboubakri** [comment]
> `Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

Applied.

**Pull request merged**